M2/Macaulay2/e/regex.cpp

```diff
@@ -0,0 +1,179 @@
+#include "engine.h"
```

Comment:
This file should be in the `d` directory, because it has nothing to do with the Groebner basis engine.  And "raw" should not be used in function names, for the same reason.  That's our convention.

Reply:
I think the `e` directory has room for other "engines" that are not GB related ... but sure.

Reply:
I don't see the problem with "raw" though. I need to separate three functions:
- rawRegexSearch in regex.cpp
- rawRegex in regex.d
- regex in regex.m2

Do you have a suggestion?

Reply:
Yes, use "regex" for all three -- those are separate name spaces, because the one in regex.d is not exported.

Here's the idea of "raw" : the type of rings has two incarnations: one at top level as the class "Ring" of top level rings, and one at C++ level as a C++ class of rings.  It would be confusing to refer to instances of both as "rings", so to distinguish them, we refer to instances of the latter as "raw rings".

When in comes to regular expressions, we have no such binary incarnation dilemma.

Reply:
> Yes, use "regex" for all three -- those are separate name spaces, whether the one in regex.d is exported or not.
> 
> Here's the idea of "raw" : the type of rings has two incarnations: one at top level as the class "Ring" of top level rings, and one at C++ level as a C++ class of rings. It would be confusing to refer to instances of both as "rings", so to distinguish them, we refer to instances of the latter as "raw rings".
> 
> When in comes to regular expressions, we have no such binary incarnation dilemma.

Oops, no, there may be a conflict between the C++ "regex" and the D "regex", if the name is not exported.  So just pick something, such as "regexSearch", for one of them.